.github/workflows/next-release.yml

```diff
@@ -0,0 +1,66 @@
+name: ⏭️ Next Release
+
+on:
+  push:
+    branches:
+      - '[0-9]+-[0-9]+'
+
+jobs:
+  next-release:
+    name: ⏭️ Next Release
+    runs-on: ubuntu-latest
+    if: github.repository_owner == 'shopify'
+    outputs:
+      NEXT_VERSION: ${{ steps.version.outputs.NEXT_VERSION }}
+    steps:
+      - name: ⬇️ Checkout repo
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: ⎔ Setup node
+        uses: actions/setup-node@v3
+        with:
+          node-version: 18
+          cache: 'npm'
+          cache-dependency-path: '**/package-lock.json'
+
+      - name: Install the packages
+        run: npm install --frozen-lockfile --ignore-engines --legacy-peer-deps
+
+      - name: 🕵️ Check for changes
+        id: version
+        run: |
+          # get latest commit sha
+          SHA=$(git rev-parse HEAD)
+          # get first 7 characters of sha
+          SHORT_SHA=${SHA::7}
+          NEXT_VERSION=next-${SHORT_SHA}
+          # set output so it can be used in other jobs
+          echo "NEXT_VERSION=${NEXT_VERSION}" >> $GITHUB_OUTPUT
+
+      - name: ⤴️ Update version
+        if: steps.version.outputs.NEXT_VERSION
+        run: |
+          git config user.email "hydrogen@shopify.com"
+          git config user.name "Hydrogen Bot"
+          git checkout -B next/${{ steps.version.outputs.NEXT_VERSION }}
+          # using changeset snapshot releases, this sets the version to 0.0.0-{tag}-DATETIMESTAMP where tag=next-SHORT_SHA
+          # as an example this results in a next release as following 0.0.0-next-1686a75-20230313113149 with a next tag
+          npm run changeset -- version --snapshot ${{steps.version.outputs.NEXT_VERSION}}
+
+      - name: 🏗 Build
+        if: steps.version.outputs.NEXT_VERSION
+        run: npm run build
+
+      - name: 🔐 Setup npm auth
+        if: steps.version.outputs.NEXT_VERSION
+        run: |
+          echo "registry=https://registry.npmjs.org" >> ~/.npmrc
+          echo "//registry.npmjs.org/:_authToken=${{ secrets.NPM_TOKEN }}" >> ~/.npmrc
+
+      # commenting until feedback
+      # - name: 🚀 Publish
+      #   if: steps.version.outputs.NEXT_VERSION
+      #   run: npm run changeset -- publish --tag next
```

Comment:
I was testing in my own repo and workflows were triggering even when in Pull requests, so I commented this out just to make sure a publish does not happen.

Reply:
Until feedback

Reply:
It will publish a next version when there is a merge to our release branch. So I don't think this is wrong.

I commented this to be safe this does not trigger even when in a PR, but I just read we should be safe:

> GitHub searches the .github/workflows directory in your repository for workflow files that are present in the associated commit SHA or Git ref of the event.